Hi @gibachan, can you take a look at the test failure?

```
/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/swift/test/expr/postfix/dot/dot_keywords.swift:104:31: error: incorrect message found
    f()   // expected-error {{static member 'f' cannot be used on instance of type 'Self'}} {{12-12=Self.}}
                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                              use of unresolved identifier 'f'
```

--- a/test/Compatibility/dot_keywords.swift
+++ b/test/Compatibility/dot_keywords.swift
@@ -1,0 +1,13 @@
+// RUN: %target-typecheck-verify-swift -swift-version 3
+
+// Static function in protocol should have `Self.` instead of its protocol name
+protocol P {}
+
+extension P {
+  static func f1() {}
+
+  func g() {
+    f1() // expected-error {{use of unresolved identifier 'f1'}}
+  }
+}
+

--- a/test/expr/postfix/dot/dot_keywords.swift
+++ b/test/expr/postfix/dot/dot_keywords.swift
@@ -101,7 +101,7 @@ extension P {
   static func f() {}
 
   func g() {
-    f()   // expected-error {{static member 'f' cannot be used on instance of type 'Self'}} {{12-12=Self.}}
+    f() // expected-error {{static member 'f' cannot be used on instance of type 'Self'}} {{5-5=Self.}}
   }
 }
 

--- a/test/expr/postfix/dot/dot_keywords.swift
+++ b/test/expr/postfix/dot/dot_keywords.swift
@@ -1,4 +1,4 @@
-// RUN: %target-typecheck-verify-swift
+// RUN: %target-typecheck-verify-swift -swift-version 4
 
 let x: Int = 1
 let y: Int = x.self

